<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgCollisionFilter</class>
 <widget class="QDialog" name="DlgCollisionFilter">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>320</width>
    <height>451</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Collision Filter</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="toolTip">
      <string/>
     </property>
     <property name="title">
      <string>This is a...</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>9</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QListView" name="categoryListView"/>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QToolButton" name="btnAddCategory">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QListWidget" name="setCategoryListWidget"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="btnRemoveCategory">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="btnRemoveAllCategories">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="toolTip">
      <string>This states the categories that this shape would accept for collision.</string>
     </property>
     <property name="title">
      <string>... and collides with</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QListView" name="maskListView"/>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QToolButton" name="btnAddMask">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QListWidget" name="setMaskListWidget"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="btnRemoveMask">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="btnRemoveAllMasks">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="toolTip">
        <string>Collision groups allow a certain group of objects to never collide (negative) or always collide (positive).
A group index of zero has no effect. Non-zero group filtering always wins against the mask bits. For example, you may want ragdolls to collide with other ragdolls but you don't want ragdoll self-collision. In this case you would give each ragdoll a unique negative group index and apply that group index to all shapes on the ragdoll.</string>
       </property>
       <property name="text">
        <string>Group Index Override</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="sbGroupIndex">
       <property name="toolTip">
        <string>Collision groups allow a certain group of objects to never collide (negative) or always collide (positive).
A group index of zero has no effect. Non-zero group filtering always wins against the mask bits. For example, you may want ragdolls to collide with other ragdolls but you don't want ragdoll self-collision. In this case you would give each ragdoll a unique negative group index and apply that group index to all shapes on the ragdoll.</string>
       </property>
       <property name="minimum">
        <number>-999999999</number>
       </property>
       <property name="maximum">
        <number>999999999</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="lblError">
       <property name="text">
        <string/>
       </property>
       <property name="textFormat">
        <enum>Qt::TextFormat::AutoText</enum>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
  <action name="actionAddCategory">
   <property name="icon">
    <iconset resource="../../../data/HyEditor.qrc">
     <normaloff>:/icons16x16/generic-right.png</normaloff>:/icons16x16/generic-right.png</iconset>
   </property>
   <property name="text">
    <string>Add Category</string>
   </property>
   <property name="toolTip">
    <string>Add selected category</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionAddMask">
   <property name="icon">
    <iconset resource="../../../data/HyEditor.qrc">
     <normaloff>:/icons16x16/generic-right.png</normaloff>:/icons16x16/generic-right.png</iconset>
   </property>
   <property name="text">
    <string>Add Mask</string>
   </property>
   <property name="toolTip">
    <string>Add selected mask to be able to collide with</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRemoveCategories">
   <property name="icon">
    <iconset resource="../../../data/HyEditor.qrc">
     <normaloff>:/icons16x16/edit-delete.png</normaloff>:/icons16x16/edit-delete.png</iconset>
   </property>
   <property name="text">
    <string>Remove Selected Categories</string>
   </property>
   <property name="toolTip">
    <string>Remove the selected categories</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRemoveMasks">
   <property name="icon">
    <iconset resource="../../../data/HyEditor.qrc">
     <normaloff>:/icons16x16/edit-delete.png</normaloff>:/icons16x16/edit-delete.png</iconset>
   </property>
   <property name="text">
    <string>Remove Selected Collision Masks</string>
   </property>
   <property name="toolTip">
    <string>Remove the selected collision mask categories</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRemoveAllCategories">
   <property name="icon">
    <iconset resource="../../../data/HyEditor.qrc">
     <normaloff>:/icons16x16/sprite-deleteAll.png</normaloff>:/icons16x16/sprite-deleteAll.png</iconset>
   </property>
   <property name="text">
    <string>Remove All Categories</string>
   </property>
   <property name="toolTip">
    <string>Clear entire categories list</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRemoveAllMasks">
   <property name="icon">
    <iconset resource="../../../data/HyEditor.qrc">
     <normaloff>:/icons16x16/sprite-deleteAll.png</normaloff>:/icons16x16/sprite-deleteAll.png</iconset>
   </property>
   <property name="text">
    <string>Remove All Masks</string>
   </property>
   <property name="toolTip">
    <string>Clear entire collision mask list</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRenameCategory">
   <property name="icon">
    <iconset resource="../../../data/HyEditor.qrc">
     <normaloff>:/icons16x16/generic-rename.png</normaloff>:/icons16x16/generic-rename.png</iconset>
   </property>
   <property name="text">
    <string>Rename Category</string>
   </property>
   <property name="toolTip">
    <string>Rename category (not used internally)</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../../../data/HyEditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DlgCollisionFilter</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DlgCollisionFilter</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
